packages/playback/src/lib/types/configuration.declarations.ts

```diff
@@ -61,10 +62,120 @@ export interface PlayerHlsConfiguration {
+
+  /**
+   * Key sessions metadata to load before starting playback.
+   * Defaults to an empty array.
+   */
+  // initialKeySessions: Array<KeySessionMetadata>;
+
+
+  // NOTE: We could make this order the priority that the user intends.
+  // If not, do we want a different config option that lists priority of key systems?
+  // Or maybe a `priority` value in the keySystems?
+  // Do we want to priortize different types of the same DRM (com.widevine.something vs com.widevine.alpha)
+  keySystems: Record<string, IKeySystemConfig>;
+
+
+  /**
+   * The time in ms to check if the media key session is expired.
+   * Defaults to 1000.
+   */
+  sessionExpirationInterval: number;
+
+  /**
+   * The minimum version of HDCP to start EME streams.
+   * Defaults to ''.
+   */
+  minHdcpVersion: string;
+
+  // This will probably replace the firstWebkitneedkeyTimeout option in contrib eme
+  /**
+   * Option to ignore duplicate init data on the encrypted event or through pssh boxes.
+   * Defaults to true
+   */
+  ignoreDuplicateInitData: boolean;
+
+
+  /**
+   * The amount of time in milliseconds to wait on the first `webkitneedkey` event before making the key request.
+   * is was implemented due to a bug in Safari where rendition switches at the start of playback can cause `webkitneedkey` to fire multiple times, with only the last one being valid.
+   * Defaults to 1000.
+   */
+  webkitneedkeyTimeout: number;
+
+  // POLYFILLS
+
+  // This would enable the legacy fairplay specific code, in a polyfill manner.
+  /**
+   * Build modern EME into browsers that use Apple's prefixed EME in Safari.
+   * Defaults to false
+   */
+  enableLegacyFairplay: boolean;
+
+  /**
+   * Adds the logic to fix setServerCertificate implementation on older platforms which claim to support modern EME.
+   * Defaults to false
+   */
+  enableSetServerCertificate: boolean;
+
+  /**
+   * Build modern EME into browsers that use legacy webkit EME API
+   * Shaka defaults this to true, I'd guess we would too.
+   */
+  enableLegacyWebkit: boolean;
+
+  /**
+   * Add support for EncryptionScheme queries in EME.
+   * https://wicg.github.io/encrypted-media-encryption-scheme/
+   * defaults to false.
+   */
+  enableEncryptionSchemes: boolean;
+
+  // keySystemsByURI: Record<string, string>;
+  // This maps something like 'urn:uuid:9a04f079-9840-4286-ab92-e65be0885f95': 'com.microsoft.playready.recommendation',
+  // I am guessing we want to do this manually. Should the user have this capability?
+  // They could just create a value in `keySystems`.
+
+  // keySystemsMapping: similar to above for two strings, just create another key session.
+
+  // delayLicenseRequestUntilPlayed?? - not sure how this would be useful
+
+  // persistentSessionOnlinePlayback?? - try playback with given persistent session ids
+  // before requesting a license. Also prevents the session removal at playback
+  // stop, as-to be able to re-use it later..
+  // DO THIS BY DEFAULT?
```

Comment:
based on your pr: is it the same as passing `initialKeySessions: Array<KeySessionMetadata>?
I would assume we should always try to reuse sessionId if we have it in the localStorage and it is still fresh

Reply:
Yeah, this is basically just be meant as a flag to enable that scenario. I guess we can assume if `initialKeySessions` is there, we want this functionality.

Reply:
I would just simplify this configuration to a boolean feature flag and implement it storing sessionIds logic internally